src/app/components/projects/add-blank-project-dialog/add-blank-project-dialog.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, OnInit } from '@angular/core';
+import { MatDialog, MatDialogRef } from '@angular/material';
+import { FormBuilder, FormGroup, Validators, FormControl } from '@angular/forms';
+import { Project } from '../../../models/project';
+import { Server } from '../../../models/server';
+import { ProjectService } from '../../../services/project.service';
+import { v4 as uuid } from 'uuid';
+import { ConfirmationDialogComponent } from '../confirmation-dialog/confirmation-dialog.component';
+import { Validator } from '../models/validator';
+import { ToasterService } from '../../../services/toaster.service';
+
+@Component({
+    selector: 'app-add-blank-project-dialog',
+    templateUrl: './add-blank-project-dialog.component.html',
+    styleUrls: ['./add-blank-project-dialog.component.css']
+})
+export class AddBlankProjectDialogComponent implements OnInit {
+    server: Server;
+    projectNameForm: FormGroup;
+
+    constructor(
+        public dialogRef: MatDialogRef<AddBlankProjectDialogComponent>,
+        private dialog: MatDialog,
+        private projectService: ProjectService,
+        private toasterService: ToasterService,
+        private formBuilder: FormBuilder) {
+            this.projectNameForm = this.formBuilder.group({
+                projectName: new FormControl(null, [Validators.required, Validator.projectNameValidator])
+            });
+        }
+
+    ngOnInit() {}
+
+    get form() {
+        return this.projectNameForm.controls;
+    }
+
+    onAddClick() : void{
+        if (!this.projectNameForm.invalid){
```

Comment:
reverse condition

Reply:
I dunno what u mean

Reply:
if (this.projectNameForm.invalid){
return;
}